crates/optimism/flashblocks/src/sequence.rs

```diff
@@ -28,13 +39,26 @@ where
     pub(crate) fn insert(&mut self, flashblock: FlashBlock) -> eyre::Result<()> {
         if flashblock.index == 0 {
             trace!(number=%flashblock.block_number(), "Tracking new flashblock sequence");
-            // Flash block at index zero resets the whole state
-            self.clear();
+            // Flash block at index zero resets the whole state.
+            let flashblocks = std::mem::take(&mut self.inner);
```

Comment:
we prefer to use `core` and `alloc` where possible over std, for no-std compat. also pls bring `core::mem` into scope in module import prelude.
```suggestion
            let flashblocks = mem::take(&mut self.inner);
```

Reply:
Done